File: old/hackerearth/golang/the-minimum-cost.go
/**
 * https://www.hackerearth.com/practice/basic-programming/bit-manipulation/basics-of-bit-manipulation/practice-problems/algorithm/min-cost-2-fe2d3308/
 **/
package main

func solveTestCaseTheMimimumCost() {
	var n, c01, c10, costA, costB int64
	scanf("\n%d %d %d\n", &n, &c01, &c10)

	bits := make([]int, n)
	for i := int64(0); i < n; i++ {
		scanf("%d", &bits[i])
	}

	a, b := 0, 1
	costA = 0
	costB = 0
	for _, bit := range bits {
		if bit != a {
			costA += int64(bit^1)*c01 + int64(bit^0)*c10
		}

		if bit != b {
			costB += int64(bit^1)*c01 + int64(bit^0)*c10
		}
		a = a ^ 1
		b = b ^ 1
	}

	ans := costB
	if costA < costB {
		ans = costA
	}
	printf("%d\n", ans)
}

func theMinimumCost() {
	var t int
	scanf("%d", &t)

	for t > 0 {
		solveTestCaseTheMimimumCost()
		t--
	}
}

File: old/hackerearth/cpp/the-minimum-cost.cpp
/**
 * https://www.hackerearth.com/practice/basic-programming/bit-manipulation/basics-of-bit-manipulation/practice-problems/algorithm/min-cost-2-fe2d3308/
 **/
#include<bits/stdc++.h>
#define lli long long int
using namespace std;

void solve_test_case() {
    lli n, c10, c01;
    cin>>n>>c01>>c10;
    vector<int> bits(n);
    for(int i=0;i<n;i++) {
        cin>>bits[i];
    }

    lli cost_a = 0;
    int a = 0;
    lli cost_b = 0;
    int b = 1;
    
    for(int i=0;i<n;i++){
        if(bits[i] != a) {
            cost_a += bits[i] == 1? c10 : c01;
        }

        if(bits[i] != b) {
            cost_b += bits[i] == 1? c10 :  c01;
        }

        a ^= 1;
        b ^= 1;
    }
    cout<< (cost_a < cost_b ? cost_a : cost_b)<<"\n";
}

int main() {
    ios_base::sync_with_stdio(false);
    int t;
    cin>>t;
    
    while(t--){
        solve_test_case();
    }
}